lang/rust/avro/src/types.rs

```diff
@@ -1323,4 +1326,236 @@ mod tests {
             JsonValue::String("936da01f-9abd-4d9d-80c7-02af85c822a8".into())
         );
     }
+
+    #[test]
+    fn test_recursive_resolves() {
```

Comment:
Please create a new ticket in JIRA and name the test cases like `test_avro_XXXX_recursive_resolves`

Reply:
Done!